File: latihan2/Mahasiswa.java
package DPBO.latihan2;

class Mahasiswa extends SivitasAkademik{
    private String NIM;
    private String fakultas;
    private String prodi;

    public Mahasiswa(){
        this.NIM = "";
        this.fakultas = "";
        this.prodi = "";
    }

    public Mahasiswa(String NIM, String fakultas, String prodi){
        super(NIM, fakultas, prodi);
        this.NIM = NIM;
        this.fakultas = fakultas;
        this.prodi = prodi;
    }

    public String getNIK(){
        return NIK;
    }

    public void setNIK(String NIK){
        this.NIK = NIK;
    }

    public String getFakultas(){
        return fakultas;
    }

    public void setFakultas(String fakultas){
        this.fakultas = fakultas;
    }

    public String getProdi(){
        return prodi;
    }

    public void setProdi(String prodi){
        this.prodi = prodi;
    }
}

File: C++/Mahasiswa.cpp
#include <iostream>
#include <string>

using namespace std;

//subclass
class Mahasiswa : public SivitasAkademik {
   private:
    string NIM;
    string fakultas;
    string prodi;

   public:
    //constructor
    Mahasiswa(string NIK, string nama, char jenis_kelamin, string asal_universitas, string email, string NIM, string fakultas, string prodi) : SivitasAkademik(NIK, nama, jenis_kelamin, asal_universitas, email) {
        this->NIM = NIM;
        this->fakultas = fakultas;
        this->prodi = prodi;
    }

    //constructor
    Mahasiswa() : SivitasAkademik() {
        this->NIM = "";
        this->fakultas = "";
        this->prodi = "";
    }

    //set NIM
    void setNIM(string NIM){
        this->NIM = NIM;
    }

    //get NIM
    string getNIM() {
        return NIM;
    }

    //set fakultas
    void setFakultas(string fakultas){
        this->fakultas = fakultas;
    }

    //get fakultas
    string getFakultas() {
        return fakultas;
    }

    //set prodi
    void setProdi(string prodi){
        this->prodi = prodi;
    }

    //get prodi
    string getProdi() {
        return prodi;
    }
};